core/src/main/java/org/apache/james/mime4j/stream/MimeConfig.java

```diff
@@ -26,6 +26,12 @@
  */
 public final class MimeConfig {
 
+    private static final MimeConfig LENIENT = MimeConfig.custom()
```

Comment:
Why private?

Reply:
Oups yes should of course be public....

Reply:
It's not only lenient (and btw you don't even call `setStringParsing(false)` ) but dangerous : max values are here to protect use from malicious input.
What happens with a mail crafted to overflow the parser ? Can I trigger JVM OutOfMemoryException with a single mail ?

Reply:
I guess so... 

Then maybe I should drop MIME4J-269, and introduce a ticket for defining in details constants to be used safely, while not failing on legitimate email, that can be used by James.

What do you think?

Reply:
Not sure to understand

Reply:
I mean, as you mention, dropping all limits to have no input limits is unsafe.

We'd better use a mime-config carefully defining a "safe enough/lenient enough" limit instead that we should be using in James.

Reply:
agree

Reply:
I propose 100 M as a safe upper bound for **maximum content length** and keep the other ones unlimited.

This should be enough to not overwhelm the JVM too much while still being very permissive.

We can call such a constant "PERMISSIVE".

WDYT?

Reply:
why not, needs some testing to check what the actual memory consumption is.